applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -191,7 +192,8 @@ static int configure_filters(void)
 	u8_t filter_mode = 0;
 	int err = configure_address_filters(&filter_mode);
 
-	if (!err && (count_bond() < CONFIG_BT_MAX_PAIRED)) {
+	if (!err && !peers_only &&
```

Comment:
We also should check if: `IS_ENABLED(CONFIG_DESKTOP_BLE_PEER_SCAN_REQUEST)`.

Reply:
Otherwise when feature is disabled we will not be able to connect with new peers when scan stops.